packages/gestalt/src/Flex.tsx

```diff
@@ -127,50 +131,63 @@ const allowedProps = [
+      gap = 0,
+      justifyContent,
+      ...rest
+    }: Props,
+    ref,
+  ) => {
+    const children = gap
+      ? // @ts-expect-error - TS2533 - Object is possibly 'null' or 'undefined'.
+        Children.map(childrenProp, (child, index) => {
+          if (child === null || child === undefined) {
+            return null;
+          }
+          return wrapWithComponent({
+            element: child,
+            Component: FlexItem,
+            props: {
+              key: index,
+            },
+          });
+        }).filter(Boolean)
+      : childrenProp;
 
-  const gapStyles = `${styles[`rowGap${typeof gap === 'number' ? gap : gap.row}`]} ${
-    styles[`columnGap${typeof gap === 'number' ? gap : gap.column}`]
-  }`;
+    const gapStyles = `${styles[`rowGap${typeof gap === 'number' ? gap : gap.row}`]} ${
+      styles[`columnGap${typeof gap === 'number' ? gap : gap.column}`]
+    }`;
 
-  const { passthroughProps, propsStyles } = buildStyles<Props>({
-    baseStyles: `${styles.Flex} ${gapStyles}`,
-    props: {
-      ...rest,
-      children,
-      direction,
-      justifyContent,
-    },
-    allowlistProps: allowedProps,
-  });
+    const { passthroughProps, propsStyles } = buildStyles<Props>({
+      baseStyles: `${styles.Flex} ${gapStyles}`,
+      props: {
+        ...rest,
+        children,
+        direction,
+        justifyContent,
+      },
+      allowlistProps: allowedProps,
+    });
+
+    // @ts-expect-error - TS2322 - Type '{ "data-test-id": string | undefined; className: string | null | undefined; style: InlineStyle | null | undefined; alignContent?: "center" | "start" | "end" | "stretch" | "between" | "around" | "evenly" | undefined; ... 18 more ...; wrap?: boolean | undefined; }' is not assignable to type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'.
+    return <div ref={ref} {...passthroughProps} {...propsStyles} data-test-id={dataTestId} />;
+  },
+);
 
-  // @ts-expect-error - TS2322 - Type '{ "data-test-id": string | undefined; className: string | null | undefined; style: InlineStyle | null | undefined; alignContent?: "center" | "start" | "end" | "stretch" | "between" | "around" | "evenly" | undefined; ... 18 more ...; wrap?: boolean | undefined; }' is not assignable to type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'.
-  return <div {...passthroughProps} {...propsStyles} data-test-id={dataTestId} />;
+// Define the type for FlexWithForwardRef to include the subcomponent, otherwise typescript does not recognize Flex.Item
+interface FlexWithSubComponents
+  extends ForwardRefExoticComponent<Props & React.RefAttributes<HTMLDivElement>> {
+  Item: typeof FlexItem;
 }
 
-Flex.Item = FlexItem;
+// Attach the subcomponent to the main component
+(FlexWithForwardRef as FlexWithSubComponents).Item = FlexItem;
+
+FlexWithForwardRef.displayName = 'Flex';
 
-Flex.displayName = 'Flex';
+export default FlexWithForwardRef as FlexWithSubComponents;
```

Comment:
try just 

`export default FlexWithForwardRef `

I tdont think you need the as FlexWithSubComponents in 193